src/cdk/app.ts

```diff
@@ -18,40 +18,29 @@ interface Prototypable {
   prototype: Stack
 }
 
-interface StackConstructable {
-  new(app: cdk.App, props: { env: string }): Stack
-}
-
-interface MiraStackConstructable {
-  new(service: MiraServiceStack): MiraStack
-}
-
-interface MiraServiceStackConstructable {
-  new(app: MiraApp): MiraServiceStack
-}
-
-export type MiraValidStack = Stack &
-  Prototypable &
-  MiraStackConstructable &
-  MiraServiceStackConstructable &
-  StackConstructable
-
+export type MiraValidStack = typeof MiraStack & typeof cdk.Stack & typeof Function
 type MiraStackList = Array<MiraValidStack>
 
 /**
  * Main Mira class.  Bootstraps CDK and loads in Stacks per user input.
  */
 export class MiraApp {
   cdkApp: cdk.App;
-  instance: MiraApp;
+  static instance: MiraApp;
   mainStack: MiraStack;
-  serviceStack: MiraServiceStack;
+  // serviceStack: MiraServiceStack;
   stackName: string;
   stacks: MiraStackList = [];
   static cliArgs = args
 
   // eslint-disable-next-line
   constructor() {
+    if (!MiraApp.instance) {
+      MiraApp.instance = this
+    } else if (args.env !== 'test') {
+      console.warn('MiraApp was instantiated twice outside a testing environment' +
```

Comment:
Why do we allow instantiation of MiraApp multiple times in `test` but not in other scenarios?

Reply:
I suppose we could, but in general it should cause errors within CDK AFAIK.